File: programming/lab2/src/main/java/moves/status/Rest.java
package moves.status;

import ru.ifmo.se.pokemon.*;

/**
 * User sleeps for 2 turns, but user is fully healed.
 */
public class Rest extends StatusMove {
    public Rest() {
        super(Type.PSYCHIC, 0, 100);
    }

    @Override
    protected boolean checkAccuracy(Pokemon pokemon, Pokemon pokemon1) {
        return true;
    }

    @Override
    protected void applySelfEffects(Pokemon pokemon) {
        pokemon.setMod(Stat.HP, (int)(pokemon.getHP() - pokemon.getStat(Stat.HP)));
        Effect restEffect = new Effect().turns(2).condition(Status.SLEEP);
        pokemon.setCondition(restEffect);
    }

    @Override
    protected String describe() {
        return "пошел отдохнуть)";
    }
}

File: programming/lab2/app/src/main/kotlin/org/itmo/moves/status/Rest.kt
package org.itmo.moves.status

import ru.ifmo.se.pokemon.*

class Rest: StatusMove(Type.PSYCHIC, 0.0, 100.0) {
    override fun checkAccuracy(p0: Pokemon?, p1: Pokemon?): Boolean {
        return true
    }

    override fun applySelfEffects(p0: Pokemon) {
        p0.setMod(Stat.HP, -(p0.getStat(Stat.HP) - p0.hp).toInt())
        p0.addEffect(Effect().turns(2).condition(Status.SLEEP))
    }

    override fun describe(): String {
        return "пошел поспать)"
    }
}